static/src/javascripts/projects/common/utils/steady-page.js

```diff
@@ -0,0 +1,188 @@
+/*
+ Utility: steady-insert.js
+ Insert components into web-pages without jumping them around by
+ moving the current scroll position by the same distance as the height of the
+ inserted element.
+ Note: Inserted elements must have a fixed height when rendered
+ (i.e: no changing the height by loading an image, lazy-loading content etc)
+ */
+
+define([
+    'common/utils/fastdom-promise',
+    'Promise',
+    'lodash/objects/assign',
+    'common/utils/Queue',
+    'common/utils/config'
+], function (
+    fastdom,
+    Promise,
+    assign,
+    Queue,
+    config
+) {
+
+    var q = new Queue();
+    var running = false;
+    var promise;
+
+    /**
+     * Given a batch and a previous currentBatchHeight, measure the height of each container
+     * in the batch
+     *
+     * @param  {Array} batch
+     */
+    function getHeightOfAllContainers (batch) {
```

Comment:
I'd move `insert` and `go` to the top of the file for readability, and put this one at the end.